### ISSUE TITLE
agent pick

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#903]
See #895 - do we need to show more stuff (like relationships) in the agent pickers? If so, what, and how should it be presented?

blocked by #732

**Comment by campmlc** [comment]
How about:

Bryan Mclean (not the same as Bryan S. Mclean)
Bryan S. Mclean (not the same as Bryan Mclean)

But we should only include not the same as relationships? Do we need "employee of" and "student of" etc?

**Comment by dustymc** [comment]
My base assumption would be that anything recorded in relationships (and other stuff) is useful in selecting the correct agent. One of the "Mclean" agents being a "student of" someone you know, or one of them was recorded as "alive" in 1634, or one having an email address of {something that makes the connection for the user}, or ... are all likely to be useful in certain situations. 

There are obvious limitations to what we can/should do in a pick. This app is run thousands of time per day, and fairly often returns hundreds of records (eg, "smith"). We (probably) do not have the computational resources to go digging through everything we know about every found agent every time they pop up. Most (some??) results are probably straightforward, and we could easily add a LOT of inefficiency by making everyone scroll through pages and pages of generally-irrelevant data every time they need to pick an agent.

There's a "more info" option, which I think could be safely expanded to just about any level of detail. I'm not sure if anyone ever clicks it or not, and I'm not sure anyone should be expected to click it for each of the 30 things that might be the "J. Smith" they're looking for.

Our current most-related agent has 10 relationships, so even that would be somewhat cumbersome to fetch and look at.

I believe most "not the same as" relationships are created only after a mess is found and fixed - it would be better to provide enough information to prevent messes in the first place. I have no idea where we might find a useful balance between doing that and being able to use the pick.

**Comment by campmlc** [comment]
How about adding the "more info" option, plus a flag to notify users and
provide a link out to potential duplicates or similar names?  That way
users would get a heads up and other options to choose between possible
duplicates without causing unnecessary info to display for unique names.

**Comment by dustymc** [comment]
>adding the "more info"

It's been there forever-ish.

>link out to potential duplicates or similar names?

I'm not sure how I'd do that. I have some scripts which attempt that (used to prevent creating potential duplicates) but they're slow and not always very accurate.

A recent influx of duplicates have inspired a new report (going out tonight, maybe) which attempts to locate agents not following https://arctosdb.org/documentation/agent/#create. Perhaps those could include "these might be the same and don't have appropriate relationships" alerts, but I would expect a lot of misses (false positives and negatives), and actually creating those relationships would be left up to an Operator. We'd still need to do something to get those data into the picks.

**Comment by dustymc** [comment]
Moving next task: need to test performance after #732

**Comment by dustymc** [comment]
I don't believe this will scale, it would make the picker very verbose, and there's a more info link available. Tentatively going wontfix and closing.

**Comment by DerekSikes** [comment]
I've thought that it would be cool to have the remarks visible next to the agents names eg '1917 Diptera expert' next to J. Smith could be VERY helpful in preventing someone from choosing J. Smith for their 2021 plant collection.

However, the remarks could be quite verbose and hard to fit nicely... thus leading to this idea: a remarks 'abstract' field. A field that holds a small number of characters only and is used to help disambiguate agents. That field could be presented in the list of agent names without taking much extra space and could save TONS of headaches downstream as we try to disentangle agents with the same names from different centuries etc.

**Comment by dustymc** [comment]
Agent pick is much more verbose in next release, feedback appreciated. (Test is currently http://arctos-test.tacc.utexas.edu)
